src/java/org/apache/cassandra/repair/RepairSession.java

```diff
@@ -372,8 +373,24 @@ public void terminate()
     public void forceShutdown(Throwable reason)
     {
         tryFailure(reason);
-        taskExecutor.shutdownNow();
+        taskExecutor.shutdown();
         terminate();
+        awaitTaskExecutorTermination();
+    }
+
+    private void awaitTaskExecutorTermination()
+    {
+        try
+        {
+            if (taskExecutor.awaitTermination(30, TimeUnit.SECONDS))
```

Comment:
What was the reasoning behind the 30 second timeout window here? Might be worth commenting on the selection of this time window.

Reply:
The only two things I've actually seen in testing are basically immediate termination and never terminating. To the extent that it identifies both of those things, it's probably fine. The interesting thing would be if abort doesn't clean things up properly, leaving a long-running streaming task open or something. 30 seconds seemed like a long enough time for us to catch that case as well. Are there any other timeouts in the repair/streaming ecosystem that would be good derivatives?